### PR TITLE
Overall update

### PR DESCRIPTION
**Pull request by userMacieG**
ox support
code cleanup + reformatted

This is not a final version.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -76,82 +91,113 @@ end
...
     licenseTable[args[2]] = false
+
     SearchedPlayer.Functions.SetMetaData("licences", licenseTable)
+
     TriggerClientEvent('ox_lib:notify', SearchedPlayer.PlayerData.source, {description = Lang:t("error.revoked_license"), type = "error"})
     TriggerClientEvent('ox_lib:notify', src, {description = Lang:t("success.revoke_license"), type = "success"})
 end)
 
-QBCore.Commands.Add("pobject", Lang:t("commands.place_object"), {{name = "type",help = Lang:t("info.poobject_object")}}, true, function(source, args)
+QBCore.Commands.Add("pobject", Lang:t("commands.place_object"), {
```
Comment (JoeSzymkowiczFiveM) [inline server/main.lua:230]: Seems like a per-line character limit is being used, to determine line breaks is that right? I don't like that certain functions' arguments are broken and some are not. Other reviewers thoughts on this are welcomed.

File: server/main.lua [hunk 2]
```diff
@@ -403,34 +501,43 @@ end)
...
-        TriggerClientEvent("inventory:client:ItemBox", src, QBCore.Shared.Items["filled_evidence_bag"], "add")
     else
         TriggerClientEvent('ox_lib:notify', src, {description = Lang:t("error.have_evidence_bag"), type = "error"})
     end
 end)
 
 RegisterNetEvent('police:server:SendTrackerLocation', function(coords, requestId)
-    local Target = QBCore.Functions.GetPlayer(source)
-    local msg = Lang:t('info.target_location', {firstname = Target.PlayerData.charinfo.firstname, lastname = Target.PlayerData.charinfo.lastname})
+    local src = source
```
Comment (JoeSzymkowiczFiveM) [inline server/main.lua:662]: Making source into a new variable is not needed here.

File: server/main.lua [hunk 3]
```diff
@@ -510,64 +632,77 @@ QBCore.Functions.CreateCallback('police:IsSilencedWeapon', function(source, cb,
...
     cb(dutyPlayers)
 end)
 
 QBCore.Functions.CreateCallback('police:GetImpoundedVehicles', function(_, cb)
     local vehicles = {}
-    MySQL.query('SELECT * FROM player_vehicles WHERE state = ?', {2}, function(result)
-        if result[1] then
+
+    MySQL.single('SELECT * FROM player_vehicles WHERE state = ?', {
+        2
```
Comment (JoeSzymkowiczFiveM) [inline server/main.lua:796]: I don't like leaving the argument floating out on a line by itself in this case.

File: server/main.lua [hunk 4]
```diff
@@ -859,48 +998,67 @@ end)
...
+                fuel,
+                plate
+            })
+
             TriggerClientEvent('ox_lib:notify', src, {description = Lang:t("info.vehicle_taken_depot", {price = price})})
         else
-            MySQL.query(
-                'UPDATE player_vehicles SET state = ?, body = ?, engine = ?, fuel = ? WHERE plate = ?',
-                {2, body, engine, fuel, plate})
+            MySQL.query('UPDATE player_vehicles SET state = ?, body = ?, engine = ?, fuel = ? WHERE plate = ?', {
```
Comment (JoeSzymkowiczFiveM) [inline server/main.lua:1260]: This seems like a waste of vertical space.

**Review by JoeSzymkowiczFiveM — COMMENTED** [review]
The code is functionally working, and just about all of my comments are around formatting, and did not touch on EVERY instance I disagreed with. If someone wants to overrule me, I would be happy to have that conversation, but I would prefer the formatting changes be reverted before merge.

File: server/main.lua (hunk 1 above)
Comment (YvanCywan) [reply server/main.lua:230]: There's also a block of code underneath, hence why I put it all on one line to begin with, to highlight the executed code underneath.

File: config.lua [hunk 5]
```diff
@@ -1,11 +1,28 @@
...
+        freeze = true
+    },
+    ["tent"] = {
+        model = joaat('prop_gazebo_03'),
+        freeze = true
+    },
+    ["light"] = {
+        model = joaat('prop_worklight_03b'),
+        freeze = true
+    }
```
Comment (YvanCywan) [inline config.lua:25]: Just why?

Nothing on those lines really needs to be changed so much that it warrants expanding it out more.

Also joaat isn't necessarily the best thing to apply here, the prop name itself is all that is needed

File: config.lua [hunk 6]
```diff
@@ -14,93 +31,84 @@ Config.HandCuffItem = 'handcuffs'
 
 Config.LicenseRank = 2
 
-Config.UseTarget = GetConvar('UseTarget', 'false') == 'true'
 Config.Locations = {
-    ["duty"] = {
```
Comment (YvanCywan) [inline config.lua:19]: These are necessary considering the way some of the code is written. Not sure why they have been changed.

File: client/job.lua [hunk 7]
```diff
@@ -374,546 +320,501 @@ RegisterNetEvent('police:client:CheckStatus', function()
...
-            minZ = v.z - 1,
-            maxZ = v.z + 1,
-        })
-    end
+    for _, v in pairs(Config.Locations["duty"]) do
+        lib.zones.box({
+            coords = v,
+            size = vec3(2, 2, 2),
+            rotation = 0.0,
+            onEnter = function(_)
```
Comment (BerkieBb) [inline client/job.lua:528]: Useless underscore

File: client/objects.lua [hunk 8]
```diff
@@ -53,80 +83,110 @@ RegisterNetEvent('police:client:spawnPObj', function(item)
...
-            lib.notify({ description = Lang:t("error.canceled"), type = "error" })
-        end
+
+    if not objectId then
+        return
+    end
+
+    if dist > 5.0 then
+        return
+    end
```
Comment (BerkieBb) [inline client/objects.lua:114]: Why 2 seperate if statements to do the same?

File: client/tracker.lua [hunk 9]
```diff
@@ -1,23 +1,34 @@
...
+            type = "error"
+        })
     end
 end)
 
 RegisterNetEvent('police:client:SetTracker', function(bool)
     local trackerClothingData = {
         outfitData = {
-            ["accessory"]   = { item = -1, texture = 0},  -- Nek / Das
+            ["accessory"] = {
```
Comment (BerkieBb) [inline client/tracker.lua:19]: If you are going to remove [""] from the variable calls, you can remove them from the creation part

File: server/main.lua [hunk 10]
```diff
@@ -30,407 +31,640 @@ local function UpdateBlips()
...
 
-QBCore.Commands.Add("grantlicense", Lang:t("commands.license_grant"), {{name = "id", help = Lang:t('info.player_id')}, {name = "license", help = Lang:t('info.license_type')}}, true, function(source, args)
-    local src = source
-    local Player = QBCore.Functions.GetPlayer(src)
+QBCore.Commands.Add("grantlicense", Lang:t("commands.license_grant"), {
+    {name = "id", help = Lang:t('info.player_id')},
+    {name = "license", help = Lang:t('info.license_type')}
+}, true, function(source, args)
+    local Player = QBCore.Functions.GetPlayer(source)
+
```
Comment (YvanCywan) [inline server/main.lua:131]: this should return to how it was before

File: server/main.lua [hunk 11]
```diff
@@ -30,407 +31,640 @@ local function UpdateBlips()
...
     if Player.PlayerData.job.type ~= "leo"  or Player.PlayerData.job.grade.level < Config.LicenseRank then
-        TriggerClientEvent('ox_lib:notify', src, {description = Lang:t("error.error_rank_license"), type = 'error'})
+        TriggerClientEvent('ox_lib:notify', source, {
+            description = Lang:t("error.error_rank_license"),
+            type = 'error'
+        })
```
Comment (YvanCywan) [inline server/main.lua:136]: same with this, and everything similar for that matter.

File: server/main.lua [hunk 12]
```diff
@@ -30,407 +31,640 @@ local function UpdateBlips()
...
         return
     end
+
     local SearchedPlayer = QBCore.Functions.GetPlayer(tonumber(args[1]))
-    if not SearchedPlayer then return end
-    local licenseTable = SearchedPlayer.PlayerData.metadata["licences"]
+
+    if not SearchedPlayer then
+        return
+    end
```
Comment (YvanCywan) [inline server/main.lua:152]: One liners that should be returned back to the way they were.

File: server/main.lua [hunk 13]
```diff
@@ -710,76 +1052,76 @@ RegisterNetEvent('police:server:JailPlayer', function(playerId, time)
...
---                         camId = camId,
---                         plate = plate,
---                         streetLabel = street1
---                     }, blipSettings)
---                 end
---             end
---         end
---     end
--- end)
+    local src = source
```
Comment (BerkieBb) [inline server/main.lua:1097]: Not necessary at all, this event does not yield anywhere

**Review by BerkieBb — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 2 above)
Comment (YvanCywan) [reply server/main.lua:662]: In all the other functions, this was removed by him, and now this is being added.

**Comment by BerkieBb** [comment]
Whilst you're at it, move the inline css from the html to the css file, that'd clean up a lot

File: server/main.lua [hunk 14]
```diff
@@ -440,20 +674,36 @@ RegisterNetEvent('police:server:SendTrackerLocation', function(coords, requestId
...
-	if args[1] then message = table.concat(args, " ") else message = Lang:t("commands.civilian_call") end
-    local ped = GetPlayerPed(src)
+QBCore.Commands.Add('911p', Lang:t("commands.police_report"), {
+    {name='message', help= Lang:t("commands.message_sent")}
+}, false, function(source, args)
+    local message = Lang:t("commands.civilian_call")
+
+	if args[1] then
+        message = table.concat(args, " ")
+    end
```
Comment (YvanCywan) [inline server/main.lua:689]:
```suggestion
	if args[1] then message = table.concat(args, " ") end
```

**Comment by BerkieBb** [comment]
For my reviews, I didn't do all cases, rather just one per 'situation'

**Review by YvanCywan — CHANGES_REQUESTED** [review]

**Comment by YvanCywan** [comment]
These are the things that bother me for the moment, while this is functional, I don't believe that the majority of changes here are necessary.

File: server/main.lua (hunk 1 above)
Comment (BerkieBb) [reply server/main.lua:230]: Rather have it on the same line

File: server/main.lua (hunk 2 above)
Comment (BerkieBb) [reply server/main.lua:662]: So you can remove it now, you only assign it to a variable if the event is yielding (a wait for example)

**Comment by Manason** [comment]
Needs to be split into smaller PRs.

**Pull request closed without merging**